### PR TITLE
Catch up to the latest Clever version of omniauth-clever

### PR DESCRIPTION
**Pull request by andrewpatterson3001**
Development on the Clever fork of our gem seems to have died, and we need to upgrade it to use a versioned v3.0/me endpoint since v1 is being shut off Dec 2022.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
 
-Unofficial OmniAuth strategy for [Clever SSO OAuth2](https://clever.com/developers/docs/oauth) integration.
+Unofficial OmniAuth strategy for [Clever SSO OAuth2](https://dev.clever.com/sso) integration.
 
 ## Installation
 
 Add the gem to your application's Gemfile:
 
-    gem 'omniauth-clever', '~> 1.1.0'
+    gem 'omniauth-clever', '~> 1.2.1'
```
Comment (zacmasinelli) [inline README.md:9]: Should this match the version listed in lib/omniauth/clever/version.rb? There is shows 1.2.2

File: lib/omniauth/strategies/clever.rb [hunk 2]
```diff
@@ -4,32 +4,43 @@
...
+      # This option bubbles up to the OmniAuth::Strategies::OAuth2
+      # when we call super in the callback_phase below.
+      # **State will still be verified** when login is initiated by the client.
+      option :provider_ignores_state, true
 
       def token_params
-        username_password = options.client_secret + ":"
         super.tap do |params|
-          params[:headers] = {'Authorization' => "Basic #{Base64.encode64(username_password)}"}
+          params[:headers] = {'Authorization' => "Basic #{Base64.strict_encode64("#{options.client_id}:#{options.client_secret}")}"}
```
Comment (zacmasinelli) [inline lib/omniauth/strategies/clever.rb:25]: Is the # still necessary/acceptable? Looking at their docs here (https://dev.clever.com/docs/identity-api - I'm assuming these are the ones I should be looking at...) it shows "Basic <clientId>:<client_secret>" without the # symbol, just want to make sure that doesn't cause any issues

Comment (andrewpatterson3001) [reply lib/omniauth/strategies/clever.rb:25]: I think that's just for string interpolation inside of the other string interpolation

File: README.md (hunk 1 above)
Comment (andrewpatterson3001) [reply README.md:9]: yeah that's correct. I'm not sure how to update that without creating a separate branch. I can do that in the next PR when we go to 2.0.0

File: lib/omniauth/strategies/clever.rb (hunk 2 above)
Comment (zacmasinelli) [reply lib/omniauth/strategies/clever.rb:25]: Ah, of course, duh. :)

**Review by zacmasinelli — APPROVED** [review]

**Pull request merged**